server/status.go

```diff
@@ -155,6 +165,65 @@ func (s *statusServer) handleLocalStatus(w http.ResponseWriter, r *http.Request,
 	w.Write(b)
 }
 
+// handleLocalLogs handles GET requests for list of available logs.
+func (s *statusServer) handleLocalLogs(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	log.Flush()
+	logFiles, err := log.ListLogFiles()
+	if err != nil {
+		log.Error(err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	b, contentType, err := util.MarshalResponse(r, logFiles, []util.EncodingType{util.JSONEncoding})
```

Comment:
I read the details of the exploit. Seems to me that just wrapping our output in an object is a decent step, but perhaps we should be requiring access to these things via POST requests? I just can't see any way that that wouldn't be an incredible pain in the ass. If everything is controlled via a single page app, we could of course do all of our requests via POST requests anyway.

I'll change all output to be wrapped in dicts.